proxy/http2/Http2Stream.cc

```diff
@@ -451,12 +451,9 @@ Http2Stream::initiating_close()
     }
     // Send EOS to let SM know that we aren't sticking around
     if (_sm && read_vio.cont) {
-      // Only bother with the EOS if we haven't sent the write complete
-      if (!sent_write_complete) {
```

Comment:
Does this mean this function sends VC_EVENT_EOS even if VC_EVENT_WRITE_COMPLETE was sent? Looks a bit weird.

Reply:
Agreed.  I was trying to back things out of the Http2 part of #6732.  Let me try again with only the h2spec timeout extension.